### DISCUSSION TITLE
Spleen 3D Segmentation - output segmentation does not invert properly

### DISCUSSION THREAD

**Discussion by park-minjun** [Project-MONAI/tutorials#1170] [General] [answered]
**Describe the bug**
After saving the output segment and opening it through a visualizer, the segmented files do not seem to inverted (both label/background and orientation)

**To Reproduce**
Run through the spleen_segmentation_3d.ipynb notebook and open the output file with the original volume in 3D Slicer (or ITKSnap, or other visualization tools)

**Expected behavior**
Segmentation should expected to have been inverted properly (from the processing transforms). Instead, the bug does not show this.

**Screenshots**
![324064903_1538106533320470_5237711145010783528_n](https://user-images.githubusercontent.com/69339164/212536527-d4ca848f-2b1d-4b2e-aa63-fcd18873c2fd.png)
![321395642_1166872593947764_2448047207592294453_n](https://user-images.githubusercontent.com/69339164/212536530-7a9e5bb7-f399-4753-b968-644c06bb3a56.png)

Hoping to get some insight in this. Thank you!

**Comment by KumoLiu** [comment]
Hi @park-minjun, may I know your MONAI version? I just tested spleen_segmentation_3d.ipynb on the latest version and visualize the output with ITKSnap, everything looks well. Could you please update your MONAI version and test the notebook again? Thanks!

**Comment by park-minjun** [comment]
> Hi @park-minjun, may I know your MONAI version? I just tested spleen_segmentation_3d.ipynb on the latest version and visualize the output with ITKSnap, everything looks well. Could you please update your MONAI version and test the notebook again? Thanks!

Hello, thank you for your response. My MONAI version is 1.1. Here are the the other config data printed with print_config():

MONAI version: 1.1.0+24.g6cb4ced0
Numpy version: 1.23.5
Pytorch version: 1.13.1+cu117
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: 6cb4ced0ea50b427a4065b4516be61b7a073abce
MONAI __file__: /workspace/clara-experiments/MONAI/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.10
Nibabel version: 5.0.0
scikit-image version: 0.19.3
Pillow version: 9.4.0
Tensorboard version: 2.11.2
gdown version: 4.6.0
TorchVision version: 0.14.1+cu117
tqdm version: 4.64.1
lmdb version: 1.4.0
psutil version: 5.9.4
pandas version: 1.5.2
einops version: 0.6.0
transformers version: 4.21.3
mlflow version: 2.1.1
pynrrd version: 1.0.0

Visualizing the output only with ITKSnap results in something like this for me, 

![image](https://user-images.githubusercontent.com/69339164/212626336-9f56a928-1e6b-49f1-b83d-98597b3a6ef7.png)

which I see is anatomically correct, but the issue arises when I try to overlay the original volume with the output segmentation.

Interestingly, attempting to open the original volume with the output segmentationdoes not seem to work as I get an error regarding segmentations having multiple components, but one component being supported by ITK Snap. This error does not occur when I open it with 3D Slicer; instead, a similar error with wrong oriented slice is shown. Do you have any pointers to what else I could look into? 

I should also note that the matplotlib looks fine when plotted, so maybe it's less of an InvertD issue but more of a SaveImaged issue? Or an error that I'm not aware of my visualizer, although I have tried both 3D Slicer / ITK Snap.

![image](https://user-images.githubusercontent.com/69339164/212629079-16b41b22-0e00-49ab-b76c-83f08539caf6.png)

**Comment by KumoLiu** [comment, marked as answer]
Hi @park-minjun, you could remove `to_onehot` in `AsDiscreted`(in `post_transforms`), then it will only save the output with one channel, then you could overlay them in the itksnap.
```
AsDiscreted(keys="pred", argmax=True, to_onehot=2)

update it to

AsDiscreted(keys="pred", argmax=True)
```

Hope it can help you, thanks!

**Comment by KumoLiu** [comment]
You could visualize it correctly is due to it only selects the foreground channel when plotting.
`plt.imshow(test_output[0].detach().cpu()[1, :, :, 20])`
